frontend/projects/upgrade/src/app/features/dashboard/feature-flags/pages/feature-flag-details-page/feature-flag-details-page-content/feature-flag-overview-details-section-card/feature-flag-overview-details-section-card.component.ts

```diff
@@ -31,22 +31,32 @@ import { Subscription } from 'rxjs';
   styleUrl: './feature-flag-overview-details-section-card.component.scss',
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class FeatureFlagOverviewDetailsSectionCardComponent {
+export class FeatureFlagOverviewDetailsSectionCardComponent implements OnInit, OnDestroy {
   isSectionCardExpanded = true;
   @Output() sectionCardExpandChange = new EventEmitter<boolean>();
+  featureFlag: FeatureFlag;
+  flagSub = new Subscription();
+  mailSub = new Subscription();
+  emailId = '';
   featureFlag$ = this.featureFlagService.selectedFeatureFlag$;
   flagOverviewDetails$ = this.featureFlagService.selectedFlagOverviewDetails;
   subscriptions = new Subscription();
-
+  confirmStatusChangeDialogRef: MatDialogRef<CommonSimpleConfirmationModalComponent>;
   menuButtonItems: IMenuButtonItem[] = [
-    { name: 'Edit', disabled: false },
-    { name: 'Delete', disabled: false },
-    { name: 'Duplicate', disabled: false },
+    { name: FEATURE_FLAG_DETAILS_PAGE_ACTIONS.EDIT, disabled: false },
+    { name: FEATURE_FLAG_DETAILS_PAGE_ACTIONS.DUPLICATE, disabled: false },
+    { name: FEATURE_FLAG_DETAILS_PAGE_ACTIONS.EXPORT_DESIGN, disabled: false },
+    { name: FEATURE_FLAG_DETAILS_PAGE_ACTIONS.EMAIL_DATA, disabled: false },
+    { name: FEATURE_FLAG_DETAILS_PAGE_ACTIONS.ARCHIVE, disabled: false },
+    { name: FEATURE_FLAG_DETAILS_PAGE_ACTIONS.DELETE, disabled: false },
   ];
 
-  confirmStatusChangeDialogRef: MatDialogRef<CommonSimpleConfirmationModalComponent>;
+  constructor(private dialogService: DialogService, private featureFlagService: FeatureFlagsService, private authService: AuthService,) {}
 
-  constructor(private dialogService: DialogService, private featureFlagService: FeatureFlagsService) {}
+  ngOnInit(): void {
+    this.flagSub = this.featureFlagService.selectedFeatureFlag$.subscribe((flag) => this.featureFlag = flag);
+    this.mailSub = this.featureFlagService.currentUserEmailAddress$.subscribe((id) => this.emailId = id);
```

Comment:
this.subscriptions.add(this.featureFlagService.currentUserEmailAddress$.subscribe((id) => this.emailId = id));